lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -3,281 +3,150 @@
+    public static async createPublicClientApplication(configuration: Configuration): Promise<IPublicClientApplication> {
 
-        this.activeSilentTokenRequests = new Map();
-        // Register listener functions
-        this.trackPageVisibility = this.trackPageVisibility.bind(this);
+        const factory = new ControllerFactory(configuration);
+        const controller = await factory.createController();
+        const pca = new PublicClientApplication(configuration, controller);
+        
+        return pca;
     }
 
-    /**
-     * Use when initiating the login process by redirecting the user's browser to the authorization endpoint. This function redirects the page, so
-     * any code that follows this function will not execute.
-     *
-     * IMPORTANT: It is NOT recommended to have code that is dependent on the resolution of the Promise. This function will navigate away from the current
-     * browser window. It currently returns a Promise in order to reflect the asynchronous nature of the code running in this function.
-     *
-     * @param request
-     */
-    async loginRedirect(request?: RedirectRequest): Promise<void> {
-        const correlationId: string = this.getRequestCorrelationId(request);
-        this.logger.verbose("loginRedirect called", correlationId);
-        return this.acquireTokenRedirect({
-            correlationId,
-            ...(request || DEFAULT_REQUEST)
-        });
+    public constructor(configuration: Configuration, controller?: IController){
+        if(controller){
+            this.controller = controller;
+        }else{
+        
+            /*
+             * If loaded in an environment where window is not available,
+             * set internal flag to false so that further requests fail.
+             * This is to support server-side rendering environments.
+             */
+            const isBrowserEnvironment = typeof window !== "undefined";
+
+            const config = buildConfiguration(configuration, isBrowserEnvironment);
+            const logger = new Logger(config.system.loggerOptions, name, version);
+            const standardOperatingContext = new StandardOperatingContext(logger, config);
+            this.controller = new StandardController(standardOperatingContext, standardOperatingContext.getConfig());
+        }
     }
 
-    /**
-     * Use when initiating the login process via opening a popup window in the user's browser
-     *
-     * @param request
-     *
-     * @returns A promise that is fulfilled when this function has completed, or rejected if an error was raised.
-     */
-    loginPopup(request?: PopupRequest): Promise<AuthenticationResult> {
-        const correlationId: string = this.getRequestCorrelationId(request);
-        this.logger.verbose("loginPopup called", correlationId);
-        return this.acquireTokenPopup({
-            correlationId,
-            ...(request || DEFAULT_REQUEST)
-        });
+    initialize(): Promise<void> {
```

Comment:
Do we want to annotate as async for consistency?

```suggestion
   async initialize(): Promise<void> {
```

Reply:
fixed.